### ISSUE TITLE
Style.css for Module wlr/workspaces does not apply properly on Hyprland

### ISSUE THREAD

**Issue by theshatterstone** [Alexays/Waybar#2121]
Hi. Thank you for waybar, and thank you for merging the persistent workspaces code. An issue I've been having for a while is with the waybar style.css for the module wlr/workspaces. The options like button.visible and button.hidden don't work on Hyprland. I decided to use the following css code to test this: 
```
#workspaces {
    padding: 0 2px;
    background: purple;
    color: #889fa7;
    border-bottom: 2px solid transparent;
}

#workspaces button {
    padding: 0 2px;
    background: yellow;
    color: #889fa7;
    border-bottom: 2px solid transparent;
}

#workspaces button.active {
    background: transparent;
    color: #dfdfdf; 
    border-bottom: 2px solid #51afef;
}

#workspaces button.focused {
    background: transparent;
    color: #dfdfdf; 
    border-bottom: 2px solid #51afef;
}

#workspaces button.visible {
    background: orange;
    color: #889fa7;
    border-bottom: 2px solid transparent;
}

#workspaces button.urgent {
    background: red;
    color: #889fa7;
    border-bottom: 2px solid transparent;
}

#workspaces button.hidden {
    background: green;
    color: #889fa7;
    border-bottom: 2px solid transparent;
}
```

The result was that the styling from `#workspaces` and `#workspaces button` apply over the styling for `button.visible`, `button.hidden`, and `button.urgent`. In fact, even when I comment out the `#workspaces` and `#workspaces button` sections, the other ones don't apply. For the version, I'm running waybar from the latest sources as of the time of writing this issue, installed using the PKGBUILD from the `waybar-hyprland-git` AUR package.

**Comment by AndreasBackx** [comment]
This is blocked by https://github.com/hyprwm/Hyprland/issues/2102 because there is no way right now that Waybar can know whether or not something is on screen or not because Hyprland does not set those values (neither does it set urgent). Once it does, it should be able to distinguish whether something is visible by looking at whether or not it's hidden. You'll have to rely on `not(:.hidden)` then until Waybar makes a change for this module that sets `.visible` if it's not hidden, which it currently does not.

**Comment by theshatterstone** [comment]
Hi there @AndreasBackx Thank you for your fast response. Just make sure I got it right: Hyprland needs to find a way to distinguish between these workspace states, and then Waybar needs to implement a way to check for these states, right? So, there's currently nothing that can be done about it, until Hyprland gets this implemented? Thank you.

**Comment by AndreasBackx** [comment]
From what I understand, yes. The protocol is not stable/merged yet and is still changing. If the workspace is not visible on the output, it should be hidden so we can then set `.visible` in Waybar. (Again, from what I understand in the protocol.)

**Comment by Barbaross93** [comment]
I'm in a similar position. I can't change the foreground color for any states of `w/r/workspaces`. Oddly though, I can change the background color and some other propoerties like padding for some of the states.

**Comment by theshatterstone** [comment]
Hello @Barbaross93 @AndreasBackx this issue has basically been fixed by #2153 and as such, I will close this issue.

**Comment by AndreasBackx** [comment]
@theshatterstone neat! I faced this when trying to move to Hyprland, though haven't made the switch as there are some bigger issues. Hoping it works, but cannot verify from my side. Though pretty sure it works. Thank you!
